examples/models/models_first_person_maze.ml

```diff
@@ -0,0 +1,143 @@
+type gui_state = {
+  camera : Raylib.Camera3D.t;
+  cubicmap : Raylib.Texture2D.t;
+  model : Raylib.Model.t;
+  texture : Raylib.Texture2D.t;
+  map_pixels : Raylib.Color.t Raylib.CArray.t;
+  map_position : Raylib.Vector3.t;
+}
+
+let gui_init () =
+  let open Raylib in
+  init_window 800 600 "Reactive GUI";
+  set_target_fps 60;
+  let camera =
+    Camera.create
+      (* Position *)
+      (Vector3.create 0.2 0.4 0.2)
+      (* Target *)
+      (Vector3.create 0.185 0.4 0.0)
+      (* Rotation over X axis *)
+      (Vector3.create 0.0 1.0 0.0)
+      (* FOV aperture in Y, degrees *)
+      45.0
+      (* Projection (perspective/orthographic) *)
+      CameraProjection.Perspective
+  in
+  (* Load cubicmap image (RAM) *)
+  let map_image = load_image "resources/cubicmap.png" in
+  let map_image_width = Image.width map_image in
+  let map_image_height = Image.height map_image in
+  (* Convert image to texture to display (VRAM) *)
+  let cubicmap = load_texture_from_image map_image in
+  let mesh = gen_mesh_cubicmap map_image (Vector3.create 1.0 1.0 1.0) in
+  let model = load_model_from_mesh mesh in
+  (* Load map texture *)
+  let texture = load_texture "resources/cubicmap_atlas.png" in
+  (* Set map diffuse texture *)
+  set_material_texture
```

Comment:
I didn't realize that this function exists. That's even better than the solution I linked above